e2e/condition_test.go

```diff
@@ -53,3 +63,212 @@ func TestCondition_CatalogServices_Include(t *testing.T) {
+	source = "../../test_modules/local_tags_file"
+	condition "catalog-services" {
+		regexp = "api-"
+		source_includes_var = true
+	}
+}
+`, taskName)
+
+	config := baseConfig().appendConsulBlock(srv).appendTerraformBlock(tempDir).
+		appendString(conditionTask)
+	configPath := filepath.Join(tempDir, configFile)
+	config.write(t, configPath)
+
+	cts, stop := ctsTestClient.StartCTS(t, configPath)
+	defer stop(t)
+
+	err := cts.WaitForAPI(15 * time.Second)
+	require.NoError(t, err)
+
+	// Test that regex filter is filtering service registration information and
+	// task triggers
+	// 0. Confirm baseline: check current number of events and that that no
+	//    catalog_service variable contains no service information
+	// 1. Register db service instance. Confirm that the task was not triggered
+	//    (no new event) and its data is filtered out of catalog_service.
+	// 2. Register api-web service instance. Confirm that task was triggered
+	//    (one new event) and its data exists in catalog_service.
+
+	// 0. Confirm only one event. Confirm empty var catalog_services
+	evenCountBase := eventCount(t, taskName, cts.Port())
+	require.Equal(t, 1, evenCountBase)
+
+	workingDir := fmt.Sprintf("%s/%s", tempDir, taskName)
+	content := testutils.CheckFile(t, true, workingDir, tftmpl.TFVarsFilename)
+	assert.Contains(t, content, "catalog_services = {\n}")
+
+	// 1. Register a filtered out service "db"
+	service := testutil.TestService{ID: "db-1", Name: "db"}
+	testutils.RegisterConsulService(t, srv, service, testutil.HealthPassing)
+	time.Sleep(7 * time.Second)
+
+	eventCountNow := eventCount(t, taskName, cts.Port())
+	require.Equal(t, evenCountBase, eventCountNow,
+		"change in event count. task was unexpectedly triggered")
+
+	content = testutils.CheckFile(t, true, workingDir, tftmpl.TFVarsFilename)
+	assert.Contains(t, content, "catalog_services = {\n}")
+
+	// 2. Register a matched service "api-web"
+	service = testutil.TestService{ID: "api-web-1", Name: "api-web"}
+	testutils.RegisterConsulService(t, srv, service, testutil.HealthPassing)
+	time.Sleep(7 * time.Second)
+
+	eventCountNow = eventCount(t, taskName, cts.Port())
+	require.Equal(t, evenCountBase+1, eventCountNow,
+		"event count did not increment once. task was not triggered as expected")
+
+	content = testutils.CheckFile(t, true, workingDir, tftmpl.TFVarsFilename)
+	assert.Contains(t, content, `"api-web" = []`)
+
+	cleanup()
+}
+
+// TestCondition_CatalogServices_NodeMeta runs the CTS binary. It specifically
+// tests a task configured with a catalog service condition with the node-meta
+// configuration set. This test confirms that when a service is registered with
+// a Consul server that does not have the configured node-meta:
+//  1) the task is not triggered (determined by confirming from the Task Status
+//     API that an event had not been added)
+//  2) the service information does not exist in the tfvars file
+func TestCondition_CatalogServices_NodeMeta(t *testing.T) {
```

Comment:
I like how you are testing the filtering functionality. It reads to me that the core portion being tested is the query params and Consul API response is filtered as expected. A (not required) improvement could be to scope down this test and change it from an e2e test to an integration test within the `templates/tftmpl/tmplfunc` package to test the query parameter `node-meta` specifically.

Reply:
I thought of writing this as an integration test but didn't want to set up the test client and went with the lazy route of writing an e2e 😅. Thanks for calling this out! I completely agree. This shouldn't be an e2e test. I'll add this on my todo list for revisit this.

Reply:
Revisited here https://github.com/hashicorp/consul-terraform-sync/pull/310!